### ISSUE TITLE
Provide CUDA requirement

### ISSUE THREAD

**Issue by fmigneault** [ai-extensions/app-package-gpu#1]
@fabricebrito 

I recommend adding `cwltool:CUDARequirement` with a `cudaDeviceCountMin` to the hints of the CWL. 

https://github.com/Duke-GCB/calrissian/blob/4ec616d0cc432f3be4a38d1eaa5f2afa049b3ccf/calrissian/job.py#L324

https://github.com/common-workflow-language/cwltool/blob/8202d5e0c05ef090438084a9210bdf778fb36370/cwltool/process.py#L969
https://github.com/common-workflow-language/cwltool/blob/8202d5e0c05ef090438084a9210bdf778fb36370/cwltool/job.py#L179

**Comment by fabricebrito** [comment]
yes @fmigneault. This comes from https://github.com/Duke-GCB/calrissian/pull/156 
We'll update this repo at some point
